File: 2022_02/Aula02/Domain/Entities/Estado.cs
namespace Aula02.Domain.Entities
{
    public class EstadoEntity
    {
        public int? EstadoID { get; private set; }
        public string UF { get; private set; }
        public string Nome { get; private set; }

        public EstadoEntity(string uf, string nome, int? estadoID=null)
        {
            UF = uf;
            Nome = nome;
            EstadoID = estadoID;
        }

        public void Update(string uf, string nome)
        {
            UF = uf;
            Nome = nome;
        }

        public override bool Equals(object? obj)
        {
            return base.Equals(EstadoID);
        }

        public override int GetHashCode()
        {
            return EstadoID.GetHashCode();
        }

        public override string ToString()
        {
            return $"[{UF}] - [{Nome}]";
        }
    }
}

File: 2022_02/aula_03_flutter/lib/features/list/data/models/estado.dart
import 'package:aula_03_flutter/features/list/domain/entities/estado.dart';
import 'package:json_annotation/json_annotation.dart';

/// flutter packages pub run build_runner watch --delete-conflicting-outputs

part 'estado.g.dart';

@JsonSerializable()
class EstadoModel extends EstadoEntity {
  const EstadoModel({
    super.estadoID,
    required super.uf,
    required super.nome,
    super.urlBandeira,
  });

  factory EstadoModel.fromJson(Map<String, dynamic> json) =>
      _$EstadoModelFromJson(json);

  /// Connect the generated [_$PersonToJson] function to the `toJson` method.
  Map<String, dynamic> toJson() => _$EstadoModelToJson(this);
}
